app/models/user.rb

```diff
@@ -5,6 +5,52 @@ class User < ApplicationRecord
   include Searchable
   include Storext.model
 
+  # NOTE: @citizen428 This is temporary code during profile migration and will
+  # be removed.
+  # rubocop:disable Metrics/BlockLength
+  concerning :ProfileMigration do
+    included do
+      PROFIE_FIELDS =
+        (Profiles::ExtractData::DIRECT_ATTRIBUTES + Profiles::ExtractData::MAPPED_ATTRIBUTES.values).freeze
+
+      default_scope { includes(:profile) } # rubocop:disable Rails/DefaultScope
+
+      # All new users should automatically have a profile
+      after_create_commit { Profile.create(user: self, data: Profiles::ExtractData.call(self)) }
```

Comment:
Rails 6 introduce more granular callbacks, which I'm using here to distinguish between creates and updates.